packages/core/FuzzedDataProvider.ts

```diff
@@ -372,33 +372,100 @@ export class FuzzedDataProvider {
 	 * is not sufficiently long.
 	 * @param maxLength the maximum length of the string
 	 * @param encoding the encoding of the string
+	 * @param printable - a boolean, which defaults to false that indicates whether consumed strings
+	 * should be forced to contain only valid printable characters
 	 * @returns a `string` of length between 0 and `maxLength` (inclusive)
 	 */
 	consumeString(
 		maxLength: number,
-		encoding: BufferEncoding | undefined = "ascii"
+		encoding: BufferEncoding | undefined = "ascii",
+		printable: boolean | undefined = false
 	): string {
 		if (maxLength < 0) throw new Error("maxLength must be non-negative");
+		let result;
 		const arrayLength = Math.min(maxLength, this._remainingBytes);
-		const result = this.data.toString(
-			encoding,
-			this.dataPtr,
-			this.dataPtr + arrayLength
-		);
+
+		if (printable) {
+			result = this.bufToPrintableString(
+				this.data,
+				this.dataPtr,
+				this.dataPtr + arrayLength,
+				encoding
+			);
+		} else {
+			result = this.data.toString(
+				encoding,
+				this.dataPtr,
+				this.dataPtr + arrayLength
+			);
+		}
 		this.dataPtr += arrayLength;
 		this._remainingBytes -= arrayLength;
 		return result;
 	}
 
+	/**
+	 * Wrapping subtraction function that returns a number within a set limit
+	 * @param min is a number representing the minuend
+	 * @param sub is a number representing the subtrahend
+	 * @param lim sets the upper boundary
+	 * @returns a number n with the constraint 0 <= n < lim, as long as lim >=2
+	 */
+	wrappingSub(min: number, sub: number, lim: number): number {
+		if (lim < 2) {
+			throw new Error("Upper boundary should be greater than or equal to 2");
+		}
+		return (min - sub + lim) % lim;
+	}
+
+	/**
+	 * A simple conversion helper that yields the ASCII representaion of a character.
+	 * @param char is a single character from a string represented as a number
+	 * @returns a 'string' from a single character that was represented by a number
+	 */
+	toAscii(char: number): number {
+		if (char < 32 || char > 126) {
+			char = (this.wrappingSub(char, 32, 255) % 95) + 32;
```

Comment:
Maybe the result can be returned without assignment.

Should be modulo 256 (instead of 255). Here is a plot with %255. Blue are uniform bytes, orange are the values mapped by `toAscii()`:
![Figure_5](https://user-images.githubusercontent.com/240414/214600931-1c1b6224-03f9-4f06-85fd-d58713acc3bf.png)

Here is a plot with %256.
![Figure_6](https://user-images.githubusercontent.com/240414/214601893-42a967ee-b6ce-42cc-bdae-39fd9b807f82.png)